be/src/exec/pipeline/exchange/exchange_sink_operator.cpp

```diff
@@ -156,52 +157,47 @@ Status ExchangeSinkOperator::Channel::send_one_chunk(const vectorized::Chunk* ch
 
 Status ExchangeSinkOperator::Channel::send_one_chunk(const vectorized::Chunk* chunk, bool eos, bool* is_real_sent) {
     *is_real_sent = false;
-
-    PTransmitChunkParams request;
-    request.set_allocated_finst_id(&_finst_id);
-    request.set_node_id(_dest_node_id);
-    request.set_sender_id(_parent->_sender_id);
-    request.set_be_number(_parent->_be_number);
+    if (_chunk_request == nullptr) {
+        _chunk_request = std::make_shared<PTransmitChunkParams>();
+        _chunk_request->set_allocated_finst_id(&_finst_id);
+        _chunk_request->set_node_id(_dest_node_id);
+        _chunk_request->set_sender_id(_parent->_sender_id);
+        _chunk_request->set_be_number(_parent->_be_number);
+    }
 
     // If chunk is not null, append it to request
     if (chunk != nullptr) {
-        auto pchunk = request.add_chunks();
+        auto pchunk = _chunk_request->add_chunks();
         RETURN_IF_ERROR(_parent->serialize_chunk(chunk, pchunk, &_is_first_chunk));
         _current_request_bytes += pchunk->data().size();
     }
 
     // Try to accumulate enough bytes before sending a RPC. When eos is true we should send
     // last packet
     if (_current_request_bytes > _parent->_request_bytes_threshold || eos) {
-        request.set_eos(eos);
-        butil::IOBuf attachment;
-        _parent->construct_brpc_attachment(&request, &attachment);
-        TransmitChunkInfo info = {this->_channel_id, std::move(request), _brpc_stub, attachment};
+        _chunk_request->set_eos(eos);
+        IOBufPtr attachment = std::make_shared<butil::IOBuf>();
+        _parent->construct_brpc_attachment(_chunk_request, attachment);
+        TransmitChunkInfo info = {this->_channel_id, _brpc_stub, std::move(_chunk_request), std::move(attachment)};
         _parent->_buffer->add_request(info);
         _current_request_bytes = 0;
-        // The original design is bad, we must release_finst_id here!
-        info.params.release_finst_id();
-
+        _chunk_request.reset();
```

Comment:
Why reset every time?

Reply:
ownership of the chunk_request has been moved to sink_buffer, calling reset explicitly only to indicate that we need create a new request next time, although std::move has alrealdy reset itself